src/builtin.c

```diff
@@ -1219,24 +1219,49 @@ static jv f_strptime(jq_state *jq, jv a, jv b) {
   memset(&tm, 0, sizeof(tm));
   const char *input = jv_string_value(a);
   const char *fmt = jv_string_value(b);
-  const char *end = strptime(input, fmt, &tm);
-
-  if (end == NULL || (*end != '\0' && !isspace(*end))) {
-    jv e = jv_invalid_with_msg(jv_string_fmt("date \"%s\" does not match format \"%s\"", input, fmt));
-    jv_free(a);
-    jv_free(b);
-    return e;
+  char *inputdup = strdup(input);
+  char *fmtdup = strdup(fmt);
+
+  int ms = -1;
+  char *pf = strstr(fmtdup, "%f");
```

Comment:
Watch out! `%` can be escaped with a preceding (unescaped) `%`.  You have to step over the format string character (well, byte) by character (byte).